src/events/components/Events.tsx

```diff
@@ -37,25 +37,41 @@ const Events = ({ location, history }: RouteComponentProps) => {
   const params = new URLSearchParams(location.search);
 
   const onTextInput = (event: ChangeEvent<HTMLInputElement>) => {
-    params.set('search', event.target.value);
+    if (event.target.value) {
+      params.set('search', event.target.value);
+    } else {
+      params.delete('search');
+    }
```

Comment:
These look quite generic?

Would probably be possible to create a function or hook which abstracts this away a bit?

Reply:
Maybe a `useQueryParam` hook?

```
const Foo = () => {
	const [startDate, changeStartDate] = useQueryParam('startDate');
	return (
		<input value={startDate} onChange={changeStartDate} ... />
	);
};

Reply:
So you to make a generic hook that sets and reads a query-parameter? But wouldn't that go against us using the queryParamContext to set the values?

Reply:
It may do.
The thought behind putting it in a context was to type all query parameters in a good way.
It may be a bit hard to work with though.

Using something like a hook to work with query parameters could make it very easy, but you would have to do all the dirty work of converting it from a string more locally.
Would also make it easier to have collisions where different part of the app uses the same key for a different type of value.

I have no idea if this works, but it is an example of a hook for query params.
It requires that we upgrade React Router to version 5.
``` typescript
export const useQueryParam = (key: string): [string | null, (value: string | null) => void] => {
  const { location, history } = useContext(__RouterContext);

  const params = new URLSearchParams(location.search);
  const value = params.get(key);

  const setValue = (newValue: string | null) => {
    if (newValue) {
      params.set(key, newValue);
    } else {
      params.delete(key);
    }
    history.replace(`${location.pathname}?${params}`);
  };

  return [value, setValue];
};
```

Reply:
But we still have the issue that the differensiert input-elements svar their value in differensiert ways, for example `<input type=«checkbox»/>` saves he state in the form of the `checked` property, and the elements `value` is pretty much useless for our use

Reply:
That is how inputs work though. It is something we have to work with (against).
It can be quite useful to create wrappers for inputs, to make them easier to use, especially things like checkboxes, radio buttons, and file inputs.

Reply:
I have now refactored it to use our new `useQueryParam`-function, would be great if you could take a look into it 😃, especially the placement of the functions, and moving them away from `Events` into `SearchModule`